mypy/newsemanal/semanal_main.py

```diff
@@ -92,13 +91,29 @@ def process_functions(graph: 'Graph', scc: List[str]) -> None:
     for module in scc:
         tree = graph[module].tree
         assert tree is not None
+        analyzer = graph[module].manager.new_semantic_analyzer
         symtable = tree.names
         targets = get_all_leaf_targets(symtable, module, None)
         for target, node, active_type in targets:
-            deferred, incomplete = semantic_analyze_target(target, graph[module], node,
-                                                           active_type)
-            assert not deferred  # There can't be cross-function forward refs
-            assert not incomplete  # Ditto
+            iteration = 0
```

Comment:
It would be better to move the new logic to a separate function. I think that `process_functions` is now too big.